### ISSUE TITLE
Mark all messages read after bulk import

### ISSUE THREAD

**Issue by swood** [mattermost/mattermost-server#9795]
Hello.

I have done massive import to Mattermost, and each client should download all imported messages and after that have to mark the all old messages as read.
I'm wondering, maybe there is a way to mark all imported messages as read? Before clients has been connected to server.

**Comment by amyblais** [comment]
Hi @swood, would the "mark_unread" field in the "ChannelNotifyProps object" be what you're looking for: https://docs.mattermost.com/deployment/bulk-loading.html#fields-of-the-channelnotifyprops-object?

**Comment by swood** [comment]
I'll try "mention" and come back if it doesn't help.

**Comment by amyblais** [comment]
Hi @swood, let me know if you were able to resolve this.

**Comment by swood** [comment]
Hello, @amyblais. Yes, it's helped for Public and Private Channels. But what could you suggest for Direct Channels?

**Comment by swood** [comment]
I've found a workaround way. Actually, I knew it when I wrote this question.
Right after you finished import you have to login as user and open all his direct channels via API-method: [view channel](https://bit.ly/2zxLRmi) (only Go-method, I tried to do it via Python and didn't rich an achievement).
After you have been "read" it, you can "close" it by mysql or pgsql request:
`update Preferences set Value = 0 where Category = "direct_channel_show"`

**Comment by swood** [comment]
Hey, @shrugalic.
I can send you a source code to read all direct channels. After you have to hide it via SQL-request.

**Comment by 7-plus-t** [comment]
Hi @swood
Could you send me or attach the code snippet here please? I've tried to mark it as read using the  REST API, but couldn't get it to work.

Thanks!

**Comment by swood** [comment]
Hey, @7-plus-t, sure:
>package main

import (
    "log"
    "os"

    mattermost "github.com/mattermost/mattermost-server/model"
)

func getClient(url string, username string, password string) *mattermost.Client4 {
    client := mattermost.NewAPIv4Client(url)
    r, e := client.Login(username, password)
    if e != nil {
        log.Fatal("Couldn't login: ", e)
    }
    log.Printf("Client logged in. Auth Token: %s.", client.AuthToken)

    log.Printf("User information: %s", r.ToJson())
    return client
}

func main() {
    url := "https://you-mattermost-server.com"
    username := os.Args[1]
    password := os.Args[2]

    client := mattermost.NewAPIv4Client(url)
    r, e := client.Login(username, password)
    log.Println(e)
    log.Printf("Client logged in. Auth Token: %s.", client.AuthToken)

    log.Printf("User information: %s", r.ToJson())

    team, _ := client.GetTeamByName("your-team-name", "")
    channels, _ := client.GetChannelsForTeamForUser(team.Id, r.Id, "")
    for _, channel := range channels {
        client.ViewChannel(r.Id, &mattermost.ChannelView{ChannelId: channel.Id, PrevChannelId: ""})
    }
}

Please be aware on an address of Mattermost server and name of a team in code. I don't know why but REST-api doesn't work for me.

**Comment by 7-plus-t** [comment]
Hi @swood,
Thanks for the code. I used it as an inspiration to retry the REST API. 
In my previous approach I tried using the /view call according to the API doc, stating that using a whitespace for the channel_id should affect all the channels of the user -> fail.
Iterating over the channels found by `/api/v4/users/%s/teams/%s/channels/members` and using them as input for `/v4/channels/members/%s/view` works!

I wrote a small Python script iterating over all the team users to mark all the channels as read and then closed them using the SQL provided by you.

Thanks for your help!
